### PR TITLE
ticdc: fix sink uri for storage sink

### PR DESCRIPTION
**Pull request by CharlesCheung96**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v7.1 (TiDB 7.1 versions)
- [x] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 1]
```diff
@@ -46,7 +46,7 @@ Amazon S3、GCS 以及 Azure Blob Storage 的 URI 参数与 BR 中这三种存
 NFS 配置样例如下：
 
 ```shell
---sink-uri="file:///my-directory/prefix"
+--sink-uri="file:///my-directory/prefix?protocol=canal-json"
```
Comment (nongfushanquan) [inline ticdc/ticdc-sink-to-cloud-storage.md:49]: Should it replace  ‘’ with '''?

**Review by db-will — APPROVED** [review]

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 2]
```diff
@@ -76,7 +98,7 @@ URI 中其他可配置的参数如下：
 {scheme}://{prefix}/{schema}/{table}/{table-version-separator}/{partition-separator}/{date-separator}/CDC{num}.{extension}
 ```
 
-- `scheme`：数据传输协议，即存储类型。例如：<code>**s3**://xxxxx</code>。
+- `scheme`：数据传输协议，存储服务类型。例如：`file`、`s3`、`gcs`、`azure`。
```
Comment (qiancai) [inline ticdc/ticdc-sink-to-cloud-storage.md:101]: 这里可以删除“数据传输协议”吗，容易和 前面的 protocol 参数混淆

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 3]
```diff
@@ -28,42 +28,64 @@ Info: {"upstream_id":7171388873935111376,"namespace":"default","id":"simple-repl
...
-
-URI 中其他可配置的参数如下：
+URI 中可配置的参数如下：
 
 | 参数              | 描述                                                   | 默认值      | 取值范围                 |
 | :--------------- | :----------------------------------------------------- | :--------- | :--------------------- |
 | `worker-count`   | 向下游存储服务保存数据变更记录的并发度                       | `16`       | `[1, 512]`             |
 | `flush-interval` | 向下游存储服务保存数据变更记录的间隔                         | `5s`       | `[2s, 10m]`            |
 | `file-size`      | 单个数据变更文件的字节数超过 `file-size` 时将其保存至存储服务中| `67108864` | `[1048576, 536870912]` |
 | `protocol`       | 输出到存储服务的消息协议                                  | N/A         | `canal-json` 和 `csv`  |
```
[the remark this answers is not in the document]
Comment (CharlesCheung96) [reply ticdc/ticdc-sink-to-cloud-storage.md:49]: 是必选配置

File: ticdc/ticdc-sink-to-cloud-storage.md (hunk 2 above)
Comment (CharlesCheung96) [reply ticdc/ticdc-sink-to-cloud-storage.md]: 已经删除了，PTAL

File: ticdc/ticdc-sink-to-cloud-storage.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (nongfushanquan) [reply ticdc/ticdc-sink-to-cloud-storage.md:49]: 必选配置是不是应该注明一下？

[the remark this answers is not in the document]
Comment (CharlesCheung96) [reply ticdc/ticdc-sink-to-cloud-storage.md:49]: 在「注意」中添加了说明，PTAL

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 4]
```diff
@@ -28,41 +28,64 @@ Info: {"upstream_id":7171388873935111376,"namespace":"default","id":"simple-repl
...
+| `enable-tidb-extension` | 可选，默认值是 `false`。输出协议为 `canal-json` 时，如果该值为 `true`，TiCDC 会发送 [Resolved 事件](/ticdc/ticdc-canal-json.md#watermark-event)，并在 Kafka 消息中添加 TiDB 扩展字段。|
 
 > **注意：**
 >
 > `flush-interval` 与 `file-size` 二者只要满足其一就会向下游写入数据变更文件。
+> 
+> `protocol` 是必选配置，如果创建 changefeed 时 cdc 无法解析到 protocol 的配置，将会返回 `CDC:ErrSinkUnknownProtocol` 错误。
```
Comment (qiancai) [inline ticdc/ticdc-sink-to-cloud-storage.md:56]:
```suggestion
> `protocol` 是必选配置，如果 TiCDC 在创建 changefeed 时未解析到该配置，将会返回 `CDC:ErrSinkUnknownProtocol` 错误。
```

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 5]
```diff
@@ -28,41 +28,64 @@ Info: {"upstream_id":7171388873935111376,"namespace":"default","id":"simple-repl
...
+--sink-uri="gcs://bucket/prefix?protocol=canal-json"
+```
+
+Azure Blob Storage 配置样例如下：
+
+```shell
+--sink-uri="azure://bucket/prefix?protocol=canal-json"
+```
+
+> **注意：**
```
Comment (qiancai) [inline ticdc/ticdc-sink-to-cloud-storage.md:78]:
```suggestion
> **建议：**
```

Comment (qiancai) [reply ticdc/ticdc-sink-to-cloud-storage.md:78]: 建议的输出效果类似这个
image

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 6]
```diff
@@ -76,7 +99,7 @@ URI 中其他可配置的参数如下：
 {scheme}://{prefix}/{schema}/{table}/{table-version-separator}/{partition-separator}/{date-separator}/CDC{num}.{extension}
 ```
 
-- `scheme`：数据传输协议，即存储类型。例如：<code>**s3**://xxxxx</code>。
+- `scheme`：存储服务类型。例如：`file`、`s3`、`gcs`、`azure`。
```
Comment (qiancai) [inline ticdc/ticdc-sink-to-cloud-storage.md:102]:
```suggestion
- `scheme`：存储服务类型。例如：`s3`、`gcs`、`azure`、`file`。
```

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 7]
```diff
@@ -28,41 +28,64 @@ Info: {"upstream_id":7171388873935111376,"namespace":"default","id":"simple-repl
...
-URI 中其他可配置的参数如下：
+URI 的 `[query_parameters]` 中可配置的参数如下：
 
 | 参数              | 描述                                                   | 默认值      | 取值范围                 |
 | :--------------- | :----------------------------------------------------- | :--------- | :--------------------- |
 | `worker-count`   | 向下游存储服务保存数据变更记录的并发度                       | `16`       | `[1, 512]`             |
 | `flush-interval` | 向下游存储服务保存数据变更记录的间隔                         | `5s`       | `[2s, 10m]`            |
 | `file-size`      | 单个数据变更文件的字节数超过 `file-size` 时将其保存至存储服务中| `67108864` | `[1048576, 536870912]` |
 | `protocol`       | 输出到存储服务的消息协议                                  | N/A         | `canal-json` 和 `csv`  |
+| `enable-tidb-extension` | 可选，默认值是 `false`。输出协议为 `canal-json` 时，如果该值为 `true`，TiCDC 会发送 [Resolved 事件](/ticdc/ticdc-canal-json.md#watermark-event)，并在 Kafka 消息中添加 TiDB 扩展字段。|
```
Comment (CharlesCheung96) [inline ticdc/ticdc-sink-to-cloud-storage.md:50]:
```suggestion
| `enable-tidb-extension` | 可选，默认值是 `false`。输出协议为 `canal-json` 时，如果该值为 `true`，TiCDC 会发送 [Resolved 事件](/ticdc/ticdc-canal-json.md#watermark-event)，并在 canal-json 消息中添加 TiDB 扩展字段。|
```

**Review by nongfushanquan — APPROVED** [review]

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 8]
```diff
@@ -28,41 +28,64 @@ Info: {"upstream_id":7171388873935111376,"namespace":"default","id":"simple-repl
...
-URI 中其他可配置的参数如下：
+URI 的 `[query_parameters]` 中可配置的参数如下：
 
 | 参数              | 描述                                                   | 默认值      | 取值范围                 |
 | :--------------- | :----------------------------------------------------- | :--------- | :--------------------- |
 | `worker-count`   | 向下游存储服务保存数据变更记录的并发度                       | `16`       | `[1, 512]`             |
 | `flush-interval` | 向下游存储服务保存数据变更记录的间隔                         | `5s`       | `[2s, 10m]`            |
 | `file-size`      | 单个数据变更文件的字节数超过 `file-size` 时将其保存至存储服务中| `67108864` | `[1048576, 536870912]` |
 | `protocol`       | 输出到存储服务的消息协议                                  | N/A         | `canal-json` 和 `csv`  |
+| `enable-tidb-extension` | 输出协议为 `canal-json` 时，如果该值为 `true`，TiCDC 会发送 [Resolved 事件](/ticdc/ticdc-canal-json.md#watermark-event)，并在 canal-json 消息中添加 TiDB 扩展字段。 | `false` | `false` 和 `true` |
```
Comment (qiancai) [inline ticdc/ticdc-sink-to-cloud-storage.md:50]:
```suggestion
| `enable-tidb-extension` | `protocol` 参数为 `canal-json` 时，如果该值为 `true`，TiCDC 会发送 [Resolved 事件](/ticdc/ticdc-canal-json.md#watermark-event)，并在 canal-json 消息中添加 [TiDB 扩展字段](/ticdc/ticdc-canal-json.md#tidb-扩展字段)。 | `false` | `false` 和 `true` |
```

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 9]
```diff
@@ -35,34 +35,59 @@ Info: {"upstream_id":7171388873935111376,"namespace":"default","id":"simple-repl
...
-URI 中其他可配置的参数如下：
+URI 的 `[query_parameters]` 中可配置的参数如下：
 
 | 参数              | 描述                                                   | 默认值      | 取值范围                 |
 | :--------------- | :----------------------------------------------------- | :--------- | :--------------------- |
 | `worker-count`   | 向下游存储服务保存数据变更记录的并发度                       | `16`       | `[1, 512]`             |
 | `flush-interval` | 向下游存储服务保存数据变更记录的间隔                         | `5s`       | `[2s, 10m]`            |
 | `file-size`      | 单个数据变更文件的字节数超过 `file-size` 时将其保存至存储服务中| `67108864` | `[1048576, 536870912]` |
 | `protocol`       | 输出到存储服务的消息协议                                  | N/A         | `canal-json` 和 `csv`  |
+| `enable-tidb-extension` | 输出协议为 `canal-json` 时，如果该值为 `true`，TiCDC 会发送 [WATERMARK 事件](/ticdc/ticdc-canal-json.md#watermark-event)，并在 canal-json 消息中添加 TiDB 扩展字段。 | `false` | `false` 和 `true` |
```
Comment (CharlesCheung96) [inline ticdc/ticdc-sink-to-cloud-storage.md:52]:
```suggestion
| `enable-tidb-extension` | `protocol` 参数为 `canal-json` 时，如果该值为 `true`，TiCDC 会发送 [WATERMARK 事件](/ticdc/ticdc-canal-json.md#watermark-event)，并在 canal-json 消息中添加 [TiDB 扩展字段](/ticdc/ticdc-canal-json.md#tidb-扩展字段)。 | `false` | `false` 和 `true` |
```

**Review by qiancai — APPROVED** [review]
LGTM

**Comment by qiancai** [comment]
Removed the `needs-cherry-pick-release-6.6` label because the parameter default value changes are not implemented in v6.6 source code

**Pull request merged**